### PR TITLE
Add TOML documentation

### PR DESCRIPTION
**Pull request by Spartee**
Add TOML interface documentation to the packages documentation. This will promote the TOML library to package modules.

### CODE REVIEW

File: modules/packages/TOML.chpl [hunk 1]
```diff
@@ -41,7 +87,8 @@ proc parseToml(input: file) : unmanaged Toml {
 }
 
 /* Receives a channel to a TOML file as a parameter and outputs an associative
```
Comment (ben-albrecht) [inline modules/packages/TOML.chpl]: Would it be more accurate to say these return a `Toml` object?

File: modules/packages/TOML.chpl [hunk 2]
```diff
@@ -21,6 +21,21 @@
 Chapel's Library for `Tom's Obvious, Minimal Language (TOML)
 <https://github.com/toml-lang/toml>`_.
 This module provides support for parsing and writing toml files.
+
+
+  .. note::
+
+    The TOML library for chapel is a work in progress. Many of the types listed in the TOML spec are not supported by this library. A core group of the spec is followed, but the library currently does not support the following types:
```
Comment (ben-albrecht) [inline modules/packages/TOML.chpl]: nitpick: spec -> [TOML specification](https://github.com/toml-lang/toml)

Hyperlink would be useful too.

**Comment by ben-albrecht** [comment]
I'd add that we're "promoting" TOML package to package modules in the PR description too.

Is there an open issue for the compiler-generated names that show up in this documentation, e.g. `AppendExpr.Call09` ?

**Comment by Spartee** [comment]
@ben-albrecht Lydia merged a fix on PR #9893

**Pull request merged**